src/main/java/de/themorpheus/edu/taskservice/controller/TaskController.java

```diff
@@ -115,4 +118,20 @@ public void deleteTask(int taskId) {
 
 		return ControllerResult.of(this.taskRepository.save(taskModel));
 	}
+
+	public ControllerResult<Object> voteTask(int taskId, int vote) {
+		TaskModel taskModel = this.taskRepository.getTaskByTaskId(taskId);
+
+		// Check if task exists
+		if (Validation.validateNull(taskModel)) return ControllerResult.of(Error.NOT_FOUND);
+
+		VotingModel votingModel = new VotingModel(
+			-1,
+			taskModel,
+			UUID.randomUUID(), // TODO: Use real user ID instead
+			vote
+		);
+		return ControllerResult.of(this.votingRepository.save(votingModel));
```

Comment:
Voting "business logic" should be separated in a new controller

Reply:
When separating the logic: don't forget to use the TaskController instead of task repository directly